test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -18,6 +19,11 @@ winrt::guid remoteId1(L"a2e4a323-b518-4799-9e80-0b37aeb0d225"); // Generated fro
 winrt::guid remoteId2(L"CA1A4AB2-AC1D-4EFC-A132-E5A191CA285A"); // Dummy guid from visual studio guid tool generator
 
 PushNotificationRegistrationToken g_appToken = nullptr;
+const wchar_t* g_bootStrapDllName = L"Microsoft.WindowsAppSDK.Bootstrap.dll";
+
+typedef HRESULT(*BootStrapTestInit)(PCWSTR prefix, PCWSTR publisherId);
+typedef HRESULT(*BootStrapInit)(const UINT32 majorMinorVersion, PCWSTR versionTag, const PACKAGE_VERSION minVersion);
+typedef void (*BootStrapShutdown)();
```

Comment:
Why?

The test\inc\ helpers know about the solution's build output layout and knows how to find the bootstrap .dll (and in the real world the WindowsAppSDK NuGet includes the dll and VS knows to copy it to dependents build output directory)

Use the test helpers in test\inc to call the bootstrap DLL and mark the Bootstrap lib as DelayLoad in your .vcxproj so your exe can load with it, then the test helpers make it accessible and voila.

Much simpler than tests replicating all this code...

Reply:
Done